examples/data_types_derive_simple.rs

```diff
@@ -0,0 +1,259 @@
+    // LowCardinality does not affect the struct definition
+    pub low_car_str: String,
+    // The same applies to a "nested" LowCardinality
+    pub nullable_low_car_str: Option<String>,
+    // FixedString is represented as raw bytes (similarly to `blob_str`, no UTF-8)
+    pub fixed_str: [u8; 16],
+    #[serde(with = "clickhouse::serde::uuid")]
+    pub uuid: uuid::Uuid,
+    #[serde(with = "clickhouse::serde::ipv4")]
+    pub ipv4: std::net::Ipv4Addr,
+    pub ipv6: std::net::Ipv6Addr,
+    pub enum8: Enum8,
+    pub enum16: Enum16,
+    pub decimal32_9_4: Decimal32,
+    pub decimal64_18_8: Decimal64,
+    pub decimal128_38_12: Decimal128,
+    #[serde(with = "clickhouse::serde::time::date")]
+    pub date: Date,
+    #[serde(with = "clickhouse::serde::time::date32")]
+    pub date32: Date,
+    #[serde(with = "clickhouse::serde::time::datetime")]
+    pub datetime: OffsetDateTime,
+    #[serde(with = "clickhouse::serde::time::datetime")]
+    pub datetime_tz: OffsetDateTime,
+    #[serde(with = "clickhouse::serde::time::datetime64::secs")]
+    pub datetime64_0: OffsetDateTime,
+    #[serde(with = "clickhouse::serde::time::datetime64::millis")]
+    pub datetime64_3: OffsetDateTime,
+    #[serde(with = "clickhouse::serde::time::datetime64::micros")]
+    pub datetime64_6: OffsetDateTime,
+    #[serde(with = "clickhouse::serde::time::datetime64::nanos")]
+    pub datetime64_9: OffsetDateTime,
+    #[serde(with = "clickhouse::serde::time::datetime64::nanos")]
+    pub datetime64_9_tz: OffsetDateTime,
+}
+
+// See ClickHouse decimal sizes: https://clickhouse.com/docs/en/sql-reference/data-types/decimal
+type Decimal32 = FixedPoint<i32, U4>; // Decimal(9, 4) = Decimal32(4)
+type Decimal64 = FixedPoint<i64, U8>; // Decimal(18, 8) = Decimal64(8)
+type Decimal128 = FixedPoint<i128, U12>; // Decimal(38, 12) = Decimal128(12)
+
+#[derive(Clone, Debug, PartialEq)]
+#[derive(serde_repr::Serialize_repr, serde_repr::Deserialize_repr)]
+#[repr(u8)]
+pub enum Enum8 {
+    Foo = 1,
+    Bar = 2,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+#[derive(serde_repr::Serialize_repr, serde_repr::Deserialize_repr)]
+#[repr(u16)]
+pub enum Enum16 {
+    Qaz = 42,
+    Qux = 255,
+}
+
+impl Row {
+    pub fn new() -> Self {
+        let mut rng = rand::thread_rng();
+        Row {
+            int8: rng.gen(),
+            int16: rng.gen(),
+            int32: rng.gen(),
+            int64: rng.gen(),
+            int128: rng.gen(),
+            uint8: rng.gen(),
+            uint16: rng.gen(),
+            uint32: rng.gen(),
+            uint64: rng.gen(),
+            uint128: rng.gen(),
+            float32: rng.gen(),
+            float64: rng.gen(),
+            boolean: rng.gen(),
+            str: random_str(),
+            blob_str: rng.gen::<[u8; 3]>().to_vec(),
+            nullable_str: Some(random_str()),
+            low_car_str: random_str(),
+            nullable_low_car_str: Some(random_str()),
+            fixed_str: rng.gen(),
```

Comment:
In general, it's fine, but not very useful actually, because usually users have `String`-like types, not arrays with specific length, so I think we should provide `#[serde(with = "clickhouse::serde::fixed_string::<20, ' '>"]` helper anyway in the future